src/librustc_typeck/check/mod.rs

```diff
@@ -4075,7 +4079,7 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                     _ => Def::Err,
                 };
                 if item_name != keywords::Invalid.name() {
-                    self.report_method_error(span, ty, item_name, None, error);
+                    self.report_method_error(span, ty, item_name, None, error, None);
```

Comment:
> If the former, then I'd prefer to output foo(...) rather than foo()

[This is the case](https://github.com/rust-lang/rust/pull/38168/files#diff-45ff4c3ed02cc75f5b8162af5d9c782bR147) now.